src/app/sidebar/sidebar.component.ts

```diff
@@ -21,24 +26,41 @@ export class SidebarComponent implements OnInit {
 
     ngOnInit() {
         this.getEvents();
-
     }
 
     toHTML(input) : any {
         return new DOMParser().parseFromString(input, "text/html").documentElement.textContent;
     }
-    
+
     getEvents(): void {
-        this.mapService.getAllEvents().subscribe(events => {
+      // console.log(this.mapService.getAllEvents().features);
+      this.mapService.getAllEvents().subscribe(events => {
         this.events = events.features;
+        this.filteredEvents = events.features;
+        this.initCategoryBar();
         for (var event of this.events) {
             this._dateService.formatDateItem(event);
         }
-    })
+      })
     }
 
-
     onSelect(event: Event): void {
-        this.selectedEvent = event;
+      this.selectedEvent = event;
     }
+
+    filterByCategory(category: string): void {
+      if (category === "all") {
+        this.filteredEvents = this.events;
+        return;
+      }
+      var tempEvents = [];
```

Comment:
If you are using es6 why use nonintuitive variable declarations, const and let are blocked scoped and are the new standard for all js dev

Reply:
We were having problems with variable declarations working with "let" on mobile friendly, so Nikhil suggested we only use let.

Reply:
There is a let in the for loop?

Reply:
Ah, our mistake.

Reply:
Also es6 let and const have full browser support http://kangax.github.io/compat-table/es6/